contracts/badger-bridge/BadgerBridgeAdapter.sol

```diff
@@ -96,57 +98,87 @@ contract BadgerBridgeAdapter is OwnableUpgradeable, ReentrancyGuardUpgradeable {
         // user args
         address _token, // either renBTC or wBTC
         uint256 _slippage,
-        address payable _destination,
+        address _user,
+        address _vault,
         // darknode args
         uint256 _amount,
         bytes32 _nHash,
         bytes calldata _sig
     ) external nonReentrant {
-        require(_token == address(renBTC) || _token == address(wBTC), "invalid token address");
+        require(_token == address(renBTC) || _token == address(wBTC) || approvedVaults[_vault], "invalid token address");
 
         // Mint renBTC tokens
-        bytes32 pHash = keccak256(abi.encode(_token, _slippage, _destination));
+        bytes32 pHash = keccak256(abi.encode(_token, _slippage, _user, _vault));
         uint256 mintAmount = registry.getGatewayBySymbol("BTC").mint(pHash, _amount, _nHash, _sig);
+
         require(mintAmount > 0, "zero mint amount");
+        
         uint256 fee = _processFee(renBTC, mintAmount, mintFeeBps);
         uint256 mintAmountMinusFee = mintAmount.sub(fee);
-
         uint256 wbtcExchanged;
+
         if (_token == address(wBTC)) {
             // Try and swap and transfer wbtc if token wbtc specified.
             uint256 startBalance = wBTC.balanceOf(address(this));
             if (_swapRenBTCForWBTC(mintAmountMinusFee, _slippage)) {
                 uint256 endBalance = wBTC.balanceOf(address(this));
                 wbtcExchanged = endBalance.sub(startBalance);
-                wBTC.safeTransfer(_destination, wbtcExchanged);
+                
+                if (_vault == address(0)) {
+                    wBTC.safeTransfer(_user, wbtcExchanged);
+                } else {
+                    wBTC.safeIncreaseAllowance(_vault, wbtcExchanged);
+                    ISett(_vault).deposit(wbtcExchanged);
+                    IERC20(_vault).safeTransfer(_user, IERC20(_vault).balanceOf(address(this)));
+                }
+
                 emit Mint(mintAmount, wbtcExchanged, fee);
                 return;
             }
         }
 
         emit Mint(mintAmount, wbtcExchanged, fee);
 
-        renBTC.safeTransfer(_destination, mintAmountMinusFee);
+        if (_vault == address(0)) {
+            renBTC.safeTransfer(_user, mintAmountMinusFee);
+        } else {
+            renBTC.safeIncreaseAllowance(_vault, mintAmountMinusFee);
+            ISett(_vault).deposit(mintAmountMinusFee);
+            IERC20(_vault).safeTransfer(_user, IERC20(_vault).balanceOf(address(this)));
+        }
```

Comment:
Let's not duplicate this logic. Let's get rid of the nested transfer/return above as well. If the swap succeeds we can prob just modify `_token` and use that below so we don't dupe logic. 

Need to check if the vault is approved. Can prob just do that check instead of `_vault == address(0)`. 

Probably should refactor out the "direct deposit" logic into it's own fn since we want to catch any reverts. The last thing we do should be the transfer of `IERC20(_token)` to the user if swap fails, deposit fails or if they originally wanted to mint renbtc.